src/ipa_pc/data_structures.rs

```diff
@@ -1,48 +1,104 @@
+//! Data structures for the dlog commitment scheme from [BCMS20]:
+//! * [`Parameters`], [`VerifierKey`], [`CommitterKey`]
+//! * [`Commitment`], [`Randomness`], [`Proof`], [`MultiPointProof`]
+//! and the implementation of their corresponding traits from [data_structures.rs](/src/data_structures.rs).
+//! In addition scheme-specific structs [`SuccinctCheckPolynomial`] and [`VerifierState`] are
+//! introduced.
+//!
+//! [BCMS20]: https://eprint.iacr.org/2020/499
 use crate::*;
 use crate::{PCCommitterKey, PCVerifierKey, Vec};
-use algebra::{AffineCurve, Field, PrimeField, ProjectiveCurve, ToBytes, UniformRand};
-use rand::thread_rng;
-use rand_core::RngCore;
+use algebra::{Curve, PrimeField};
 use std::{
     convert::TryFrom,
     io::{Read, Write},
     vec,
 };
 
-/// `UniversalParams` are the universal parameters for the inner product arg scheme.
+// #[cfg(test)]
+use rand::thread_rng;
+// #[cfg(test)]
+// use algebra::UniformRand;
+
+/// `Parameters` are the polynomial commitment parameters for the inner product arg scheme.
+// TODO: it seems to be artificial to have a `Parameters` struct which is the same as
+// the committer key. Let us investigate if this struct is really needed.
+// (Can't we simply implement the PCParameters trait for the struct PCCommitterKey?)
```

Comment:
This was left more for backward compatibility. We can remove PCParameters and Parameters, in this case function "setup" in PC will return keys pair directly, right? Something like this:

setup(max_degree: usize) -> Result<(Self::CommitterKey, Self::VerifierKey), Self::Error>;

Reply:
BTW what the purpose of trim here if we remove the parameters struct? Now we can have the parameters with longer comm_key and trimmed to the segment_size committer / verifier keys. After removing when we will use trim and for what? If setup function returns keys, they can be already generated for the needed size

Reply:
Not really. Consider a situation in which we have many circuits for which creating the proofs, but of different sizes (like in our CCTP) so that we want to use different segment sizes. In this case we will generate only one set of DLOG keys, with a certain size, and trim them according to the situation. Given that, I'm afraid we cannot remove this struct (otherwise when we trim we will lose the other generators, unless we re-generate them but it seems a bit of a waste)

Reply:
In regards of the inline comment, I think it is possible to drop the separate Parameters struct, and implement `PCParameters` for the `CommitterKey` alone.

Reply:
I've done this in this way - implemented PCParameters directly for CommitterKey, but to be honest don't like the resulted structure. Take a look please

Reply:
Well setup and setup_from_seed functions should still output (CommitterKey, VerifierKey). Currently is not generic enough (keys are the same only for DLOG. For Kate, for instance, it's not like this)

Reply:
Ok I restored setup and setup_from_seed functions (that now return (CommitterKey, VerifierKey)).
I dropped the PCParameters trait in favor of a PCKey trait containing only degree, get_hash, trim functions.
I dropped the PCCommitterKey and PCVerifierKey traits (that were containing the exact same stuff) in favor of a PCKey trait, and I requested types CommitterKey and VerifierKey in PolynomialCommitment trait to be of that type.